src/app/client/src/app/modules/public/components/signup/signup.component.spec.ts

```diff
@@ -2,31 +2,128 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+    component.ngOnInit();
+    expect(component.signUpForm.valid).toBeFalsy();
+  });
+  it('should show validation error message for userName', () => {
+    component.ngOnInit();
+    let errors = {};
+    const userName = component.signUpForm.controls['userName'];
+    errors = userName.errors || {};
+    expect(errors['required']).toBeTruthy();
+  });
+  it('should show pattern match error message for userName', () => {
+    component.ngOnInit();
+    let errors = {};
+    const userName = component.signUpForm.controls['userName'];
+    userName.setValue('test');
+    errors = userName.errors || {};
+    expect(errors['pattern']).toBeTruthy();
+  });
+  it('should show validation error message for firstName', () => {
+    component.ngOnInit();
+    let errors = {};
+    const firstName = component.signUpForm.controls['firstName'];
+    errors = firstName.errors || {};
+    expect(errors['required']).toBeTruthy();
+  });
+  it('should show pattern match error message for firstName', () => {
+    component.ngOnInit();
+    let errors = {};
+    const firstName = component.signUpForm.controls['firstName'];
+    firstName.setValue('    test  ');
+    errors = firstName.errors || {};
+    expect(errors['pattern']).toBeTruthy();
+  });
+  it('should show validation error message for phone', () => {
+    component.ngOnInit();
+    let errors = {};
+    const phone = component.signUpForm.controls['phone'];
+    errors = phone.errors || {};
+    expect(errors['required']).toBeTruthy();
+  });
+  it('should show pattern match error message for phone', () => {
+    component.ngOnInit();
+    let errors = {};
+    const phone = component.signUpForm.controls['phone'];
+    phone.setValue('8989');
+    errors = phone.errors || {};
+    expect(errors['pattern']).toBeTruthy();
+  });
+  it('email field validity', () => {
+    component.ngOnInit();
+    let errors = {};
+    const email = component.signUpForm.controls['email'];
+    errors = email.errors || {};
+    expect(errors['required']).toBeTruthy();
+  });
+  it('should show pattern match error message for email', () => {
+    component.ngOnInit();
+    let errors = {};
+    const email = component.signUpForm.controls['email'];
+    email.setValue('test');
+    errors = email.errors || {};
+    expect(errors['pattern']).toBeTruthy();
+  });
+  it('should call onSubmitForm method and return success response', () => {
+    const router = TestBed.get(Router);
+    component.signUpForm = new FormGroup({});
+    const resourceService = TestBed.get(ResourceService);
+    resourceService.messages = mockSignUpResponse.resourceBundle.messages;
+    const signupService = TestBed.get(SignupService);
+    spyOn(signupService, 'signup').and.callFake(() => Observable.of(mockSignUpResponse.successResponse));
+    component.onSubmitForm();
+    expect(router.navigate).toHaveBeenCalledWith(['']);
+    expect(mockSignUpResponse.resourceBundle.messages.smsg.m0039).toBe('Sign up successfully, Please login...');
```

Comment:
what is this assertion, remove this. Check toaster service tohave been called with that message.